### PR TITLE
Restore style checking on travis

### PR DESCRIPTION
**Pull request by paulb777**
Travis style checking was broken in #3020. This PR does a proper fix. See tests in pb-tmp* in recent runs at https://travis-ci.org/firebase/firebase-ios-sdk/builds. 

It may be helpful to see the delta from before #3020 ` git diff bad992edb3cbb4295d216941cb2571961b2a1da0 -- scripts/check.sh`

### CODE REVIEW

File: Firebase/Core/FIRApp.m [hunk 1]
```diff
@@ -91,7 +91,7 @@ @interface FIRApp ()
 
 @end
 
-@implementation FIRApp
+@implementation   FIRApp
```
Comment (paulb777) [inline Firebase/Core/FIRApp.m]: Will be reverted after confirmation that the travis PR run catches this.

**Comment by paulb777** [comment]
Style checking restoration confirmed in https://travis-ci.org/firebase/firebase-ios-sdk/jobs/536972856

File: scripts/check.sh [hunk 2]
```diff
@@ -165,17 +165,30 @@ fi
...
-START_SHA=$(git rev-parse "${START_REVISION}")
+  NEW_RANGE_START=$(git merge-base origin/master "${RANGE_END}")
+  START_REVISION="${START_REVISION/$RANGE_START/$NEW_RANGE_START}"
+
+  START_SHA="${START_REVISION}"
+
+elif ! git rev-parse origin/master >& /dev/null; then
+  git remote set-branches --add origin master
+  git fetch origin
+  START_SHA=$(git merge-base origin/master "${TRAVIS_COMMIT}")
```
Comment (wilhuff) [inline scripts/check.sh]: I don't understand the logic here.

If we're given a range, we handle it as above. If not given a range, why are we checking origin/master?

Comment (paulb777) [reply scripts/check.sh]: To compute START_SHA when $TRAVIS_COMMIT_RANGE is empty.  See https://travis-ci.org/firebase/firebase-ios-sdk/jobs/536944398

Comment (wilhuff) [reply scripts/check.sh]: I understand the effect you're going for.

What I'm saying is that reading the code, it doesn't follow that if we're not processing a commit range that we're necessarily checking against `origin/master` and that `TRAVIS_COMMIT` would necessarily be set.

In particular, this could have been run by a user, and they pass a single commit that they want to check against, in which case it's irrelevant whether or not `origin/master` is available in the repo. Indeed the change you've made is harmful to this usage because `TRAVIS_COMMIT` is unset in that case.

All told I think we should cleanly separate the concerns of figuring out what we're checking from any repair of it. This change mixes these up in a way that's confusing and ultimately wrong for some use cases.

I recommend the following changes:
  * Change the travis configuration to not pass $TRAVIS_COMMIT_RANGE. This makes it unambiguous that travis passes arguments via environment variable.
  * Before parsing arguments, try to determine `START_REVISION` from travis environment variables
    * Something like `if [[ -n "${TRAVIS_COMMIT_RANGE:-}" ]]` would be the test to see if the variable is set and non-empty without violating `-o nounset`.
    * Check for both `TRAVIS_COMMIT_RANGE` and `TRAVIS_COMMIT`
    * In either case, all you do is assign to `START_REVISION`
  * Then down here:
    * Handle the range case essentially as we do, because travis can give busted ranges
    * If `START_REVISION` starts with `origin`, ensure that it's actually in the repo

File: scripts/check.sh [hunk 3]
```diff
@@ -150,6 +151,22 @@ if [[ "${ALLOW_DIRTY}" == true && "${COMMIT_METHOD}" == "message" ]]; then
   exit 1
 fi
 
+# When travis clones a repo for building, it uses a shallow clone. After the
+# first commit on a non-master branch, TRAVIS_COMMIT_RANGE is not set, master
+# is not available and we need to compute the START_REVISION from the common
+# ancestor of $TRAVIS_COMMIT and origin/master.
+if [[ "${TEST_ONLY}" == true ]]; then
```
Comment (wilhuff) [inline scripts/check.sh]: `test-only` only indicates we want no changes made, not that it's necessarily a travis run. I think this is unrelated.

File: scripts/check.sh [hunk 4]
```diff
@@ -150,6 +151,22 @@ if [[ "${ALLOW_DIRTY}" == true && "${COMMIT_METHOD}" == "message" ]]; then
...
+  if ! [[ -z $TRAVIS_COMMIT_RANGE ]] ; then
```
Comment (wilhuff) [inline scripts/check.sh]: This doesn't work when running under `set -u`, which fails when you access an unset variable. You need to default an unset variable to empty explicitly with a `:-` in braces, as in `${TRAVIS_COMMIT_RANGE:-}`.

`! [[ -z foo ]]` is more idiomatically written as `[[ -n foo ]]`.

File: scripts/check.sh [hunk 5]
```diff
@@ -150,6 +151,22 @@ if [[ "${ALLOW_DIRTY}" == true && "${COMMIT_METHOD}" == "message" ]]; then
...
+    START_REVISION="$TRAVIS_COMMIT_RANGE"
+  elif ! [[ -z $TRAVIS_COMMIT ]] ; then
+    if ! git rev-parse origin/master >& /dev/null; then
```
Comment (wilhuff) [inline scripts/check.sh]: This is unrelated to whether or not `TRAVIS_COMMIT` is set, isn't it?

`origin/master` only mattered specifically previously because it was the default for `START_REVISION`.

File: scripts/check.sh [hunk 6]
```diff
@@ -163,29 +180,9 @@ if [[ "${VERBOSE}" == true ]]; then
   env | egrep '^TRAVIS_(BRANCH|COMMIT|PULL|REPO)' | sort || true
 fi
 
-# When travis clones a repo for building, it uses a shallow clone. After the
-# first commit on a non-master branch, TRAVIS_COMMIT_RANGE is not set and
-# START_REVISION is "origin/master" because a range wasn't passed in.
 if [[ "${START_REVISION}" == *..* ]]; then
-  RANGE_START="${START_REVISION/..*/}"
```
Comment (wilhuff) [inline scripts/check.sh]: I don't think this should be removed, or alternatively, this needs to be in the block where you handle `TRAVIS_COMMIT_RANGE`, if you specifically want to handle travis separately.

This was fixing a problem where travis would give us a commit range where the first commit was on some other branch and that's no longer being fixed in here.

**Comment by paulb777** [comment]
@wilhuff All comments should now be addressed. Thanks!
